Packetfilter API: add support for IPv6

tpantelis opened the pull request:
Modified the `DriverFn` and `packetfilter.New` functions to take the IP family.

tpantelis committed:
Packetfilter API: add support for IPv6

Modified the DriverFn and packetfilter.New functions to take the IP family.

Signed-off-by: Tom Pantelis <tompantelis@gmail.com>

yboaron commented on pkg/packetfilter/nftables/nftables.go:
```diff
@@ -79,7 +80,11 @@ type packetFilter struct {
 
-func New() (packetfilter.Driver, error) {
+func New(family k8snet.IPFamily) (packetfilter.Driver, error) {
+	if family != k8snet.IPv4 {
```
Maybe we can also enable  IPV6 driver creation for nftables, it should be safe because currently all clients will only create a V4 driver

tpantelis committed:
Remove Family field from SetInfo

This is not needed since the underyling packetfilter Driver
implementation knows what IP family it represents.

Signed-off-by: Tom Pantelis <tompantelis@gmail.com>

tpantelis commented on pkg/packetfilter/nftables/namedset.go:
```diff
@@ -35,6 +36,9 @@ type namedSet struct {
 func (p *packetFilter) NewNamedSet(set *packetfilter.SetInfo) packetfilter.NamedSet {
 	setType := "ipv4_addr"
+	if p.family == k8snet.IPv6 {
+		setType = "ipv6_addr"
```
@yboaron I assume this is correct for IPv6? I couldn't find any concrete info/doc to confirm it...

tpantelis commented:
@yboaron I [updated the PR](https://github.com/submariner-io/submariner/compare/71822a74216a4e8a283e9d23976ddf93a9778be8..54899922041fd3c76337169b2a81432f9d7d24ce) to add verification of the expected IP family of IPs/CIDRs passed via the API (eg Rules) in the fake packet filter implementation and to [remove the Family field from SetInfo](https://github.com/submariner-io/submariner/pull/3350/commits/54899922041fd3c76337169b2a81432f9d7d24ce).
